common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -40,14 +42,22 @@
 public final class ThreadDeathWatcher {
 
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(ThreadDeathWatcher.class);
-    private static final ThreadFactory threadFactory =
-            new DefaultThreadFactory(ThreadDeathWatcher.class, true, Thread.MIN_PRIORITY);
+    private static final ThreadFactory threadFactory;
 
     private static final Queue<Entry> pendingEntries = PlatformDependent.newMpscQueue();
     private static final Watcher watcher = new Watcher();
     private static final AtomicBoolean started = new AtomicBoolean();
     private static volatile Thread watcherThread;
 
+    static {
+        String poolName = "threadDeathWatcher";
+        String serviceThreadPrefix = SystemPropertyUtil.get("io.netty.serviceThreadPrefix");
+        if (!StringUtil.isNullOrEmpty(serviceThreadPrefix)) {
+            poolName = serviceThreadPrefix + ' ' + poolName;
```

Comment:
'  ' space is used in IntelliJ Platform — "ApplicationImpl pooled thread #n", "FS Synchroniser", "Alarm pool(shared)" and so on.

So, it seems, we should not automatically add any separator here to avoid such questions, right? Another solution — give complete freedom to format thread name, I think, is complicated and is not necessary.